src/pages/api/donation-settings/[name].ts

```diff
@@ -0,0 +1,44 @@
+import { PrismaClientKnownRequestError } from "@prisma/client/runtime";
+import { NextApiHandler, NextApiRequest, NextApiResponse } from "next";
+import { getDonationSettings } from "~/lib/db/donationSettings";
+
+const handler: NextApiHandler = async (req, res) => {
+  switch (req.method) {
+    case "GET":
+      getStreamerDonationSettings(req, res);
+      break;
+    default:
+      res.setHeader("Allow", ["GET"]);
+      res.status(405).end(`Method ${req.method} Not Allowed`);
+  }
+};
+
+const getStreamerDonationSettings = async (
+  request: NextApiRequest,
+  response: NextApiResponse
+) => {
+  const { name } = request.query;
+
+  if (!name) throw new Error("Provide a streamer name");
+
+  try {
+    const donationSettings = await getDonationSettings(String(name));
+
+    if (donationSettings) {
+      response.status(200).json({ data: donationSettings });
+      return;
+    }
+
+    throw new Error("This user does not exist");
+  } catch (error) {
+    if (error instanceof PrismaClientKnownRequestError) {
```

Comment:
Leak prisma internals into http handler? We should consider a stricter interface for these kind of things.